components/writers/src/main/java/org/datacleaner/extension/output/CreateExcelSpreadsheetAnalyzer.java

```diff
@@ -94,14 +111,31 @@ public void configureForTransformedData(AnalysisJobBuilder ajb, TransformerDescr
 
     @Override
     public OutputWriter createOutputWriter() {
-
-        if (file.exists() && overwriteFileIfExists) {
-            final String pathname = file.getPath();
-            final boolean delete = file.delete();
-            if (delete) {
-                file = new File(pathname);
-            } else {
-                throw new IllegalStateException("The existing file could not be deleted");
+        if (file.exists()) {
+            ExcelDatastore datastore = new ExcelDatastore(file.getName(), new FileResource(file),
+                    file.getAbsolutePath());
+            try (final UpdateableDatastoreConnection connection = datastore.openConnection()) {
+                final DataContext dataContext = connection.getDataContext();
+                final Schema[] schemas = dataContext.getSchemas();
+                if (schemas.length >= 1) {
+                    final String[] tableNames = schemas[1].getTableNames();
```

Comment:
Better way to get the proper schema is to say dataContext.getDefaultSchema() ... taking schemas[1] is actually not guaranteed to work since schemas are sorted alphabetically and then you rely on the filename (which is used as schema name) to be sorted after "information_schema" :-)